### PR TITLE
feat: HTTP Trigger

### PR DESCRIPTION
**Pull request by loicmathieu**
Fixes https://github.com/kestra-io/kestra/issues/2299

Flow to test the new trigger:
```yaml
id: http-trigger
namespace: dev

triggers:
  - id: http
    type: io.kestra.plugin.fs.http.Trigger
    uri: https://api.chucknorris.io/jokes/random
    responseCondition: "{{ response.statusCode == 200 }}"

tasks:
  - id: hello
    type: io.kestra.core.tasks.log.Log
    message: Kestra team wishes you a great day! 👋
```

### CODE REVIEW

**Comment by anna-geller** [comment]
why response instead of trigger.statusCode? just thinking whether this is not confusing where the response variable is coming from as it's usually trigger.XYZ

**Review by anna-geller — APPROVED** [review]
forget my previous comment, this makes sense, the trigger variables can be used in tasks - sorry, my misinterpretation

works!

```yaml
id: http-trigger
namespace: dev


tasks:
  - id: hello
    type: io.kestra.core.tasks.log.Log
    message: "{{trigger.body}}"

triggers:
  - id: http
    type: io.kestra.plugin.fs.http.Trigger
    uri: https://api.chucknorris.io/jokes/random
    responseCondition: "{{ response.statusCode == 200 }}"
    interval: PT30S
```

**Comment by loicmathieu** [comment]
The trigger variables are normally to use in the triggered Flow tasks as you noticed, here we evaluate the trigger.
So I decided to call it response as it's a variable only usable when evaluating the trigger property responseCondition.

However, I can rename it.

I can also totally remove the responseCondition property and only rely on the standard conditions property where you can use a condition of type VariableCondition. But I found it more logical to define the evaluation condition like this. The responseCondition will be evaluated prior to any standard trigger conditions.

File: src/main/java/io/kestra/plugin/fs/http/AbstractHttp.java [hunk 1]
```diff
@@ -44,56 +44,27 @@
 @EqualsAndHashCode
 @Getter
 @NoArgsConstructor
-abstract public class AbstractHttp extends Task {
-    @Schema(
-        title = "The fully-qualified URI that points to the HTTP destination"
-    )
-    @PluginProperty(dynamic = true)
+abstract public class AbstractHttp extends Task implements HttpInterface{
```
Comment (Skraye) [inline src/main/java/io/kestra/plugin/fs/http/AbstractHttp.java:47]:
```suggestion
abstract public class AbstractHttp extends Task implements HttpInterface {
```

**Review by Skraye — APPROVED** [review]
Small indent detail, however LGTM

**Pull request merged**